android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -1908,6 +1932,25 @@ private void onPostNotificationPermissionResult(boolean granted)
       Logger.w(TAG, "Permission POST_NOTIFICATIONS has been refused");
   }
 
+  private void onUnrestrictedBatteryPermissionResult(ActivityResult result)
+  {
+    if (result.getResultCode() == Activity.RESULT_OK)
+      Logger.i(TAG, "A request to disable battery optimization has been approved");
+    else
+      Logger.w(TAG, "A request to disable battery optimization has been refused");
+
+    if (mUnrestrictedBatteryPermissionRequestedBy == null)
+      return;
+    Config.setNeedToRequestUnrestrictedBatteryPermission(mUnrestrictedBatteryPermissionRequestedBy, false);
```

Comment:
1. Why it should be saved to the config instead of checking the state dynamically?
2. Why false if user has refused it?

Reply:
> 1. Why it should be saved to the config instead of checking the state dynamically?
Otherwise it will ask permissions everytime when user will turn on the setting and annoy them. Other applications like starva and mapmyride also ask only once. 
> 2. Why false if user has refused it?
To keep it stored that user don't want to change the settings and we should not annoy them again and again for it.